ldms/src/core/ldms_stream.c

```diff
@@ -478,6 +491,7 @@ __stream_deliver(struct ldms_addr *src, uint64_t msg_gn,
 			.name_len = name_len,
 			.data_len = data_len,
 			.name = stream_name,
+			.name_hash = hash,
```

Comment:
If you're going to put the hash in the message, you could put the entire thing in there and avoid the computation on every message. The port and addr aren't changing. This would also mean that the same hash is used all the way up the chain from sender to consumer.

Reply:
The address got changed in the following case.

There is a logic to exclude the use of `local` address (e.g. `127.0.0.1`) as `src` address here:

https://github.com/ovis-hpc/ovis/blob/f71e7adcff4a852f3ccc05b143a7794a6bc71986/ldms/src/core/ldms_stream.c#L1131-L1153

This is to prevent the case that the application running on the same host connects to sampler ldmsd using `localhost` address and make all `src` in the messages from all nodes being `127.0.0.1`. Basically, if the `src` address is local, do not resolve it and let the next guy resolve it instead.

Reply:
So there must be code somewhere else that checks the address and family and if 0 sets it to the address and port of the forwarder? Where is that logic?

Reply:
@narategithub In any event, all we're doing here is randomizing the seed to what we hope will be a random hash. So as-usual, we've found ourselves in a hole so we keep digging. Let's just back up and decide how we're going to generate a random hash.